### PR TITLE
HDDS-7750. Incorrect WRITE ACL check

### PR DESCRIPTION
**Pull request by aryangupta1998**
## What changes were proposed in this pull request?

Changing checkKeyAcls() to checkKeyAclsInOpenKeyTable() in S3MultipartUploadCommitPartRequest as for Acl type "WRITE", the key can only be found in OpenKeyTable.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-7750

## How was this patch tested?

Tested Manually

### CODE REVIEW

File: hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/s3/multipart/TestS3MultipartRequest.java [hunk 1]
```diff
@@ -83,6 +84,8 @@ public void setup() throws Exception {
     when(ozoneManager.getMetrics()).thenReturn(omMetrics);
     when(ozoneManager.getMetadataManager()).thenReturn(omMetadataManager);
     auditLogger = Mockito.mock(AuditLogger.class);
+    OmMetadataReader omMetadataReader = Mockito.mock(OmMetadataReader.class);
+    when(ozoneManager.getOmMetadataReader()).thenReturn(omMetadataReader);
```
Comment (Xushaohong) [inline hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/s3/multipart/TestS3MultipartRequest.java:88]: This line is duplicated with line 85. Need to verify if the mock is necessary.
`when(ozoneManager.getMetadataManager()).thenReturn(omMetadataManager);`

Comment (adoroszlai) [reply hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/s3/multipart/TestS3MultipartRequest.java:88]: They seem to be different: `getMetadataManager` vs. `getOmMetadataReader`.

Comment (Xushaohong) [reply hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/s3/multipart/TestS3MultipartRequest.java:88]: Oops, my mistake.

**Review by sadanand48 — COMMENTED** [review]
Thanks @aryangupta1998 for the patch, change looks good to me. The exact issue described in the jira seems not to be an issue.

**Comment by adoroszlai** [comment]
@aryangupta1998 @sadanand48 I have reverted this from `master` because compilation is failing:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:testCompile (default-testCompile) on project ozone-manager: Compilation failure
[ERROR] /home/runner/work/ozone/ozone/hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/s3/multipart/TestS3MultipartRequest.java:[89,45] no suitable method found for thenReturn(org.apache.hadoop.ozone.om.OmMetadataReader)
[ERROR]     method org.mockito.stubbing.OngoingStubbing.thenReturn(org.apache.hadoop.ozone.om.snapshot.ReferenceCounted<org.apache.hadoop.ozone.om.IOmMetadataReader,org.apache.hadoop.ozone.om.snapshot.SnapshotCache>) is not applicable
[ERROR]       (argument mismatch; org.apache.hadoop.ozone.om.OmMetadataReader cannot be converted to org.apache.hadoop.ozone.om.snapshot.ReferenceCounted<org.apache.hadoop.ozone.om.IOmMetadataReader,org.apache.hadoop.ozone.om.snapshot.SnapshotCache>)
[ERROR]     method org.mockito.stubbing.OngoingStubbing.thenReturn(org.apache.hadoop.ozone.om.snapshot.ReferenceCounted<org.apache.hadoop.ozone.om.IOmMetadataReader,org.apache.hadoop.ozone.om.snapshot.SnapshotCache>,org.apache.hadoop.ozone.om.snapshot.ReferenceCounted<org.apache.hadoop.ozone.om.IOmMetadataReader,org.apache.hadoop.ozone.om.snapshot.SnapshotCache>...) is not applicable
[ERROR]       (argument mismatch; org.apache.hadoop.ozone.om.OmMetadataReader cannot be converted to org.apache.hadoop.ozone.om.snapshot.ReferenceCounted<org.apache.hadoop.ozone.om.IOmMetadataReader,org.apache.hadoop.ozone.om.snapshot.SnapshotCache>)
```

#4567 changed `OzoneManager`:

```diff
-  public OmMetadataReader getOmMetadataReader() {
-    return omMetadataReader;
+  public ReferenceCounted<
+      IOmMetadataReader, SnapshotCache> getOmMetadataReader() {
+    return rcOmMetadataReader;
```

**Pull request merged**